wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wom/graph/renaming/package.scala

```diff
@@ -0,0 +1,111 @@
+package wdl.draft3.transforms.wdlom2wom.graph
+
+import wdl.draft3.transforms.wdlom2wom.expression.renaming.IdentifierLookupRenamer
+import wdl.model.draft3.elements._
+import wdl.draft3.transforms.wdlom2wom.graph.renaming.GraphIdentifierLookupRenamer.ops._
+import wdl.draft3.transforms.wdlom2wom.expression.renaming.IdentifierLookupRenamer.ops._
+
+package object renaming {
+
+  implicit val workflowDefinitionIdentifierRenamer: GraphIdentifierLookupRenamer[WorkflowDefinitionElement] = new GraphIdentifierLookupRenamer[WorkflowDefinitionElement] {
+    override def renameIdentifiers(a: WorkflowDefinitionElement,
+                                   renamingMap: Map[String, String])
+                                  (implicit expressionElementRenamer: IdentifierLookupRenamer[ExpressionElement],
+                                   graphIdentifierLookupRenamer: GraphIdentifierLookupRenamer[WorkflowGraphElement]): WorkflowDefinitionElement = {
+      val newInputsSection = a.inputsSection map { inputSection =>
+        inputSection.copy(inputDeclarations = inputSection.inputDeclarations.map {
+          id =>
+            val renamingMapWithoutThisValue = renamingMap.filterNot(_._1 == id.name)
+            id.copy(expression = id.expression.map(_.renameIdentifiers(renamingMapWithoutThisValue)))
+        })
+      }
+
+      val newGraphElements = a.graphElements map { ge => ge.renameIdentifiers(renamingMap) }
+
+      val newOutputsSection = a.outputsSection map { outputSection =>
+        outputSection.copy(outputs = outputSection.outputs.map {
+          od => od.copy(expression = od.expression.renameIdentifiers(renamingMap))
+        })
+      }
+
+
+      a.copy(inputsSection = newInputsSection,
+        graphElements = newGraphElements,
+        outputsSection = newOutputsSection)
+    }
+  }
+
+  implicit val graphElementIdentifierRenamer = new GraphIdentifierLookupRenamer[WorkflowGraphElement] {
+    override def renameIdentifiers(a: WorkflowGraphElement,
+                                   renamingMap: Map[String, String])
+                                  (implicit expressionElementRenamer: IdentifierLookupRenamer[ExpressionElement],
+                                   graphIdentifierLookupRenamer: GraphIdentifierLookupRenamer[WorkflowGraphElement]): WorkflowGraphElement = a match {
```

Comment:
Yeah exactly, it's choosing which of the subfunctions to call based on type, I think it just looks weird because they're all called the same thing and have the same signature